org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/IScannableDeviceService.java

```diff
@@ -100,34 +103,30 @@
 	 * objects. DeviceInformation is JSON serializable and this method is 
 	 * 
 	 * @return
+	 * @throws ScanningException 
 	 * @throws Exception
 	 */
-	default Collection<DeviceInformation<?>> getDeviceInformation() throws Exception {
+	default Collection<DeviceInformation<?>> getDeviceInformation() throws ScanningException {
 
 		final Collection<String> names = getScannableNames();
 		final Collection<DeviceInformation<?>> ret = new ArrayList<>(names.size());
 		for (String name : names) {
-	
-			IScannable<?> device = getScannable(name);
-			if (device==null) throw new ScanningException("There is no created device called '"+name+"'");
-	
-			DeviceInformation<?> info = new DeviceInformation<Object>(name);
-			Util.merge(info, device);
-			ret.add(info);
+			try {
+				final IScannable<?> device = getScannable(name);
+				if (device == null) {
+					throw new ScanningException("There is no created device called '" + name + "'");
+				}
+				ret.add(((AbstractScannable<?>) device).getDeviceInformation());
+			} catch (Exception e) {
+			}
 		}
 		return ret;
 	}
+	
+	public default void handleDeviceError(String name, Exception e) {
+		System.err.println("Failure getting device information for " + name);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)